### ISSUE TITLE
Container Definition Environment variables are being removed when empty

### ISSUE THREAD

**Issue by wichon** [aws-actions/amazon-ecs-deploy-task-definition#216]
# Problem statement
This action is removing environment variables with empty values inside the container definition.

# What is the bug/problem behavior?
When deploying a new task definition version, the deployment process is removing container definition Environment variables which have empty values, causing the deploy to potentially fail due to the missing Environment variables.

This is present in aws-actions/amazon-ecs-deploy-task-definition@v1

# What is the desired/correct behavior
Environment variables inside the container definition should remain unmodified.

# Steps to reproduce
- Create a dummy ECS service with a set of environment variables, and leave one empty.
- Deploy a task definition with this action
- New deployed task definition is missing the environment variable that is empty.

# Notes
I tested this with the [render task definition action](https://github.com/aws-actions/amazon-ecs-render-task-definition) and downloading the task definition with the cli using the [describe-task-defintion](https://docs.aws.amazon.com/cli/latest/reference/ecs/describe-task-definition.html) command and I can confirm that just before the task definition is deployed, the Environment variables with empty values are still in the task definition json file.

# Possible implementation details
Perhaps checking that the resulting task definition has the critical user provided settings unchanged (things like cpu, mem, Environment variables, volumes mapping, etc)

# Workarounds
We need to manually add back the missing environment variable which will make our service stay in an inconsistent state while the missing environment variable is added.

**Comment by wichon** [comment]
@paragbhingre Thanks for your time investigating this issue, I bet there is a lot going on your side, do you have any news on why its happening?

**Comment by paragbhingre** [comment]
@wichon Thank you for your patience, would you mind sharing me your task definition json? I would like to see how you are sending your parameters in the environments and how they are getting empty. You can just share me before and after getting empty task def json.

**Comment by wichon** [comment]
@paragbhingre Sure, I am attaching the before and after task defs


[ecs-deploy-issue-task-defs.zip](https://github.com/aws-actions/amazon-ecs-deploy-task-definition/files/6651469/ecs-deploy-issue-task-defs.zip)

**Comment by paragbhingre** [comment]
Could you please confirm if environment variables in task def json looks life following snippet when you check them in github action logs

```
"environment": [
                {
                   "name":"DEMO_VAR"
                }
                {
                    "name": "APP_PLACEHOLDER_SERVICE",
                    "value": "ecs-demo-site"
                },
                {
                    "name": "APP_PLACEHOLDER_ENV",
                    "value": "qa"
                },
                {
                    "name": "APP_PLACEHOLDER_PORT",
                    "value": "80"
                }
            ], 
```

Just the `value` param is missing and not the whole object ?

**Comment by wichon** [comment]
Here is how that block looks like when I download the task def:
```
"environment": [
        {
          "name": "DEMO_VAR",
          "value": ""
        },
        {
          "name": "APP_PLACEHOLDER_SERVICE",
          "value": "ecs-demo-site"
        },
        {
          "name": "APP_PLACEHOLDER_ENV",
          "value": "qa"
        },
        {
          "name": "APP_PLACEHOLDER_PORT",
          "value": "80"
        }
    ]
```

And I am using this command to download the task def:
aws ecs describe-task-definition --task-definition $AWS_ECS_TASK_DEFINITION_FAMILY --query taskDefinition > task-definition.json

**Comment by allisaurus** [comment]
This should be fixed in [1.4.7](https://github.com/aws-actions/amazon-ecs-deploy-task-definition/releases/tag/v1.4.7)
